Cython/Compiler/ExprNodes.py

```diff
@@ -13313,11 +13502,13 @@ class CoerceToBooleanNode(CoercionNode):
         Builtin.bytes_type:      'PyBytes_GET_SIZE',
         Builtin.bytearray_type:  'PyByteArray_GET_SIZE',
         Builtin.unicode_type:    '__Pyx_PyUnicode_IS_TRUE',
+        PyrexTypes.FastcallTupleType(): '__Pyx_FastcallTuple_Len',
+        PyrexTypes.FastcallDictType(): '__Pyx_FastcallDict_Len'
```

Comment:
Can these types be kept around as constants, like other types at the end of `PyrexTypes.py`?

Reply:
I'm currently using individual instances for `FastcallTupleType` and `FastcallDictType` for each function because they're the best place to count the number of coercions. Therefore the usual `type is PyrexTypes.singleton_instance` test won't work.

I think it is worth counting the coercions - if there's too many then it becomes a pretty pointless optimization and it's better to do it once at the start of the function. However this might be something I try to revise since I don't really like the way that the fallback is done.